subsys/settings/src/settings_store.c

```diff
@@ -48,8 +50,15 @@ void settings_dst_register(struct settings_store *cs)
 static void settings_load_cb(char *name, void *val_read_cb_ctx, off_t off,
 			     void *cb_arg)
 {
-	 int rc = settings_set_value_priv(name, val_read_cb_ctx, off, 0);
-	__ASSERT(rc == 0, "set-value operation failure\n");
+	int rc = settings_set_value_priv(name, val_read_cb_ctx, off, 0);
+
+	if (rc != 0) {
+		LOG_ERR("set-value failure. key: %s error(%d)",
+			name, rc);
+	} else {
+		LOG_INF("set-value OK. key: %s",
```

Comment:
I think this LOG_INF is too much information. It could be on DEBUG level or even removed.